admin/tool/policy/acceptances.php

```diff
@@ -28,15 +28,73 @@
 $policyid = optional_param('policyid', null, PARAM_INT);
 $versionid = optional_param('versionid', null, PARAM_INT);
 $userid = optional_param('userid', null, PARAM_INT);
+$acceptfor = optional_param('acceptfor', null, PARAM_INT);
 
-admin_externalpage_setup('tool_policy_managedocs', '', ['policyid' => $policyid, 'versionid' => $versionid],
-    new moodle_url('/admin/tool/policy/acceptances.php',
-        ['policyid' => $policyid, 'versionid' => $versionid, 'userid' => $userid]));
-require_capability('tool/policy:viewacceptances', \context_system::instance());
+if ($acceptfor) {
+    $versionid = required_param('versionid', PARAM_INT);
+    $usercontext = context_user::instance($acceptfor);
+    require_capability('tool/policy:acceptbehalf', $usercontext);
+    $userid = $acceptfor;
+}
 
-$output = $PAGE->get_renderer('tool_policy');
+// Set up the page either in an individual user context or as an admin page 'tool_policy_managedocs'.
+if ($userid || !has_capability('tool/policy:viewacceptances', context_system::instance())) {
+    // Viewing report for the individual user - either for oneself or for mentee or this is a privacy officer.
+    require_login();
+    if (isguestuser()) {
+        print_error('noguest');
+    }
+    $context = context_user::instance($userid ?: $USER->id);
+    if ($userid && $USER->id != $userid) {
+        if (!has_any_capability(['tool/policy:acceptbehalf', 'tool/policy:viewacceptances'], $context)) {
+            require_capability('tool/policy:viewacceptances', $context);
+        }
+    } else {
+        $userid = $USER->id;
+    }
+    $PAGE->set_context($context);
+    $PAGE->set_url(new moodle_url('/admin/tool/policy/acceptances.php', ['userid' => $userid]));
+} else {
+    // Viewing report for multiple users.
+    admin_externalpage_setup('tool_policy_managedocs', '', ['policyid' => $policyid, 'versionid' => $versionid],
+        new moodle_url('/admin/tool/policy/acceptances.php', ['policyid' => $policyid, 'versionid' => $versionid]));
+}
+
+$policy = null;
+if ($versionid) {
+    $policy = tool_policy\api::get_policy_version($policyid, $versionid);
+} else if ($policyid) {
+    $policy = tool_policy\api::get_policy($policyid);
+    $policy->policyid = $policy->id;
+}
+
+if ($policy && !$userid) {
+    $PAGE->navbar->add(format_string($policy->name),
+        new moodle_url('/admin/tool/policy/managedocs.php', ['id' => $policy->policyid]));
+}
 $PAGE->navbar->add(get_string('useracceptances', 'tool_policy'));
 
+if ($acceptfor) {
+    $user = $DB->get_record('user', ['id' => $acceptfor], 'id,'.get_all_user_name_fields(true), MUST_EXIST);
+    $returnurl = optional_param('returnurl', null, PARAM_LOCALURL);
```

Comment:
I generally prefer if all params the page accepts are enlisted together at the start of the script, implicitly self-documenting the script's interface.